target_chains/sui/sdk/js/src/SuiPriceServiceConnection.ts

```diff
@@ -0,0 +1,21 @@
+import {
+  PriceServiceConnection,
+  HexString,
+} from "@pythnetwork/price-service-client";
+import { Buffer } from "buffer";
+
+export class SuiPriceServiceConnection extends PriceServiceConnection {
+  /**
+   * Gets price update data (either batch price attestation VAAs or accumulator messages, depending on the chosen endpoint), which then
+   * can be submitted to the Pyth contract to update the prices. This will throw an axios error if there is a network problem or
+   * the price service returns a non-ok response (e.g: Invalid price ids)
+   *
+   * @param priceIds Array of hex-encoded price ids.
+   * @returns Array of price update data.
+   */
+  async getPriceFeedsUpdateData(priceIds: HexString[]): Promise<number[][]> {
```

Comment:
`number[]` is a weird type. I prefer passing arrays of buffers instead and do the conversion on the edges as Buffer is more known as the data blob type.